sharpy/linear/src/libss.py

```diff
@@ -148,6 +156,89 @@ def states(self):
     def states(self, value):
         self._states = value
 
+    @property
```

Comment:
for outputs, states etc. you call the setter within the getter. Not sure how rigorous this is: if you were to do instance.states = x, but instance.A.shape[0] was a different number, when retrieving instance.states you would get A.shape[0] not what you set to x... 
So in this case I would remove the setter altogether and just do  self._states = self.A.shape[0] in the getter. 
Minor thing anyway